cirq/value/states.py

```diff
@@ -0,0 +1,365 @@
+        return '+X'
+
+    def __repr__(self):
+        return 'cirq.PLUS_X'
+
+    def state_vector(self):
+        return np.array([1, 1]) / np.sqrt(2)
+
+    def stabilized_by(self):
+        from cirq import ops
+        return 1, ops.X
+
+
+class _MinusX(_Named1QState):
+    def __str__(self):
+        return '-X'
+
+    def __repr__(self):
+        return 'cirq.MINUS_X'
+
+    def state_vector(self):
+        return np.array([1, -1]) / np.sqrt(2)
+
+    def stabilized_by(self):
+        from cirq import ops
+        return -1, ops.X
+
+
+class _PlusY(_Named1QState):
+    def __str__(self):
+        return '+Y'
+
+    def __repr__(self):
+        return 'cirq.PLUS_Y'
+
+    def state_vector(self):
+        return np.array([1, 1j]) / np.sqrt(2)
+
+    def stabilized_by(self):
+        from cirq import ops
+        return 1, ops.Y
+
+
+class _MinusY(_Named1QState):
+
+    def __str__(self):
+        return '-Y'
+
+    def __repr__(self):
+        return 'cirq.MINUS_Y'
+
+    def state_vector(self):
+        return np.array([1, -1j]) / np.sqrt(2)
+
+    def stabilized_by(self):
+        from cirq import ops
+        return -1, ops.Y
+
+
+class _PlusZ(_Named1QState):
+    def __str__(self):
+        return '+Z'
+
+    def __repr__(self):
+        return 'cirq.PLUS_Z'
+
+    def state_vector(self):
+        return np.array([1, 0])
+
+    def stabilized_by(self):
+        from cirq import ops
+        return 1, ops.Z
+
+
+class _MinusZ(_Named1QState):
+    def __str__(self):
+        return '-Z'
+
+    def __repr__(self):
+        return 'cirq.MINUS_Z'
+
+    def state_vector(self):
+        return np.array([0, 1])
+
+    def stabilized_by(self):
+        from cirq import ops
+        return -1, ops.Z
+
+
+PLUS_X = _PlusX()
+document(
+    PLUS_X, """The |+> State
+    
+    This is the state such that X|+> = +1 |+>
+
+    Vector:
+
+        [1, 1] / sqrt(2)
+    """)
+
+MINUS_X = _MinusX()
+document(
+    MINUS_X, """The |-> State
+    
+    This is the state such that X|-> = -1 |->
+
+    Vector:
+
+        [1, -1] / sqrt(2)
+    """)
+
+PLUS_Y = _PlusY()
+document(
+    PLUS_Y, """The |i> State
+    
+    This is the state such that Y|i> = +1 |i>
+
+    Vector:
+
+        [1, i] / sqrt(2)
+    """)
+
+MINUS_Y = _MinusY()
+document(
+    MINUS_Y, """The |-i> State
+
+    This is the state such that Y|-i> = -1 |-i>
+
+    Vector:
+
+        [1, -i] / sqrt(2)
+    """)
+
+PLUS_Z = _PlusZ()
+document(
+    PLUS_Z, """The |0> State
+
+    This is the state such that Z|0> = +1 |0>
+
+    Vector:
+
+        [1, 0]
+    """)
+
+MINUS_Z = _MinusZ()
+document(
+    MINUS_Z, """The |1> State
+
+    This is the state such that Z|1> = -1 |1>
+
+    Vector:
+
+        [0, 1]
+    """)
+
+PAULI_STATES = [
+    PLUS_X,
+    MINUS_X,
+    PLUS_Y,
+    MINUS_Y,
+    PLUS_Z,
+    MINUS_Z,
+]
+document(PAULI_STATES,
+         """All one-qubit states stabalized by the pauli operators.""")
+
+
+def zeros_state(qubits: Iterable['cirq.Qid']) -> TensorProductState:
```

Comment:
Is there a context where you would do this instead of using `initial_state=0`?

Reply:
removing this for now